src/client.ts

```diff
@@ -233,16 +242,16 @@ export class Client {
    * due to an internal error.
    */
   async query<T extends QueryValue>(
-    request: Query,
-    headers?: QueryRequestHeaders
+    query: Query,
+    options?: QueryOptions
   ): Promise<QuerySuccess<T>> {
     if (this.#isClosed) {
       throw new ClientClosedError(
         "Your client is closed. No further requests can be issued."
       );
     }
 
-    return this.#query(request.toQuery(headers));
+    return this.#query(query, options);
```

Comment:
i don't think we're ready to make this change yet. 

I suspect this is why we regressed to a `ClientError` from a `TypeError`.

Let's wait on this until we can do this without regressing errors we fire on bad inputs.

Reply:
Didn't change it all the way back, but did move the `toQuery` up out of `#query`.